text/2019-08-21-supported-platforms.md

```diff
@@ -0,0 +1,230 @@
+  targets. TiKV *should probably* build on these platforms (through official or
+  ad-hoc tooling), and we'll contributor accept patches to fix or improve these
+  builds. Users are strongly discouraged from using these targets for any
+  purpose other than R&D.
+      + Example: NixOS, a Linux distro with a slightly exotic packaging concept
+      that none of our maintainers have experience with
+      + Not recommended
+      + No instructions provided
+      + No binaries/containers provided
+* **Not supported:** Testing has found the target not viable in the past. These
+  targets are either not possible to build TiKV on, or known to have serious
+  problems. Maintainers may accept patches to add or improve support if they are
+  uninvasive, but open an issue first. *Please, don't use these.*
+      + Example: Windows, we have never managed to build or successfully run
+        tests on this OS
+      + Not recommended
+      + No instructions provided
+      + No support available, bug reports about this platform may be closed
+        without reason
+
+### Target Tier/Platform Promotion
+
+Tier promotion on the **development platform** is possible via an RFC. An RFC of
+this nature should be accompanied by a series of of PRs adding content depending
+on the tier promotion. Tier promotion on the **deployment platform** is
+determined by maintainer consensus.
+
+Development Platform Promotions:
+
+* **Not supported to Tier 3:**
+      + The target can successfully run `cargo check`, `cargo clippy`, `rls`, and
+        `cargo fmt` on TiKV without patches or modifications
+      + The target can successfully run `cargo build` and `cargo build --release`
+        on TiKV without patches or modifications
+      + The target can produce development builds of TiKV, they do not
+        necessarily run
+* **Tier 3 to Tier 2:**
+      + The target can host and run TiKV as a service, and serve requests
+      + The target can actively particpate in a TiKV cluster containing TiKV
+        nodes running on tier 1 targets
+      + The target can pass TiKV's test suite and benchmark suite
+      + The proposal is submitted by a regular contributor to TiKV, and they
+        state they intend to maintain support
+* **Tier 2 to Tier 1:**
+      + The target has reliably passed TiKVs test and benchmark suites for a
+        period of 3 months
+      + The target can use an interactive debugger on TiKV and documentation is present
+      + The target supports at least one fuzzer and fuzzing has been performed
+      + The target is used as a development machine by at least one maintainer
+        or three regular contributors
+
+### Tier Definitions
+
+This RFC proposes the following definitions for the tiers of platforms and targets:
+
+* **Deployment Platforms**
+      + **Tier 1**:
+        - CentOS 7.x.x on x86-64 on x84-64 with SSE4.2
+        - Ubuntu 16.04 through 19.04 on x84-64 with SSE4.2
```

Comment:
AIUI, this adding official support, are we comfortable doing it?

Reply:
@c4pt0r do you know?